pkg/router/controller/controller.go

```diff
@@ -97,7 +114,21 @@ func (c *RouterController) HandleEndpoints() {
 	c.lock.Lock()
 	defer c.lock.Unlock()
 
+	// Change the local sync state within the lock to ensure that all
+	// event handlers have the same view of sync state.
+	c.endpointsReplaceRead = c.EndpointsReplaceRead()
```

Comment:
> I'm assuming you meant 'just not seeing'. A sync from etcd never results in router state being cleared. The new stream of events are used to modify the existing router state.

Looking in from the outside, that seems a bit unusual.  I normally think of a resync as an opportunity to rebuild from scratch just in case something in the current state got wonked.  If you're never going to change the router state, why both relisting or resyncing at all?